addons/website_profile/controllers/main.py

```diff
@@ -269,7 +269,7 @@ def _get_position_map(self, position_domain, users, group_by):
             position_map = self._get_user_tracking_karma_gain_position(position_domain, users.ids, group_by)
         else:
             position_results = users._get_karma_position(position_domain)
-            position_map = dict((user_data['user_id'], dict(user_data)) for user_data in position_results)
+            position_map = {user_data['user_id']: dict(user_data) for user_data in position_results}
```

Comment:
Same here, these functions return distinct objects already, and they are not used below.
```suggestion
            position_map = {user_data['user_id']: user_data for user_data in position_results}
```